js/app/prepare-data/index.js

```diff
@@ -0,0 +1,18 @@
+import fetcher from "../../lib/fetcher.js";
+import prepareItemsData from "./items.js";
+import prepareItemData from "./item.js";
+
+export default async function({ baseURI, resource }) {
+
+    // fetch the raw data
+    const dataURI = `${baseURI}${resource}`;
```

Comment:
Moving this comment to the right place:

`export default async function prepareData(pageOptions)`

could you then use it as:

`pageOptions.baseURI` ?

Reply:
Because javascript isn't compiled (until it runs), there are fewer safeguards that e.g. one has not typed a variable name incorrectly. I think this is probably why the javascript community try not to type the name of objects more than they need to. However, this is probably a historical influence as modern IDEs are pretty good about pointing out undeclared variables in javascript.

Can you have a read at this to further inform our conversation https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/ ?

Reply:
> Can you have a read at this to further inform our conversation https://hacks.mozilla.org/2015/05/es6-in-depth-destructuring/ ?

I can certainly see some very useful applications of this, but in our particular example I am still not convinced it achieves better clarity or considerable improvement to readability. But, I am happy to accept your argument and keep it.

Reply:
I've not made an argument either way - I'm still trying to understand your perspective.

Reply:
shall we do a slack/teams call to discuss this when convenient - might cover it more quickly that way

Reply:
> shall we do a slack/teams call to discuss this when convenient - might cover it more quickly that way

Yes, message me to let me know when it suits you.

Reply:
Per offline discussion, we've agree that the destructuring vs options object is a matter of choice and style rather than particular benefit